src/mvc/collection.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Collections are ordered sets of models.
+ *
+ * Any event that is triggered on a model in a collection will also be triggered on the collection directly, for
+ * convenience.
+ *
+ * @class Collection
+ */
+
+CKEDITOR.define( [ 'emitter', 'utils' ], function( EmitterMixin, utils ) {
+	function Collection() {
+		/**
+		 * The internal list of models in the collection.
+		 *
+		 * @property _models
+		 * @private
+		 */
+		Object.defineProperty( this, '_models', {
+			value: []
+		} );
+
+		/**
+		 * The number of items available in the collection.
+		 *
+		 * @property length
+		 */
+		Object.defineProperty( this, 'length', {
+			get: function() {
+				return this._models.length;
+			}
+		} );
+	}
+
+	/**
+	 * @inheritdoc utils#extend
+	 */
+	Collection.extend = utils.extendMixin;
+
+	utils.extend( Collection.prototype, EmitterMixin, {
+		/**
+		 * Adds an item into the collection.
+		 *
+		 * Note that this is an array-like collection, so the same item can be present more than once. This behavior is
+		 * for performance purposes only and is not guaranteed to be kept in the same way in the future.
+		 *
+		 * @param {Model} model The item to be added.
+		 */
+		add: function( model ) {
+			model.addParentEmitter( this );
```

Comment:
I was unsure about this as well, but it brought two benefits:

 1. It makes it much easier to have such event bubbling system implemented in parent objects. It's just a matter of calling a single function, instead of listening to an event and having to fire it again (which in any case is still possible to be done).

 2. It performs better because it is executed inside the `fire()` method itself through a simple loop over the parents instead of using the events API machinery.

 3. Then, the critical part... listenTo what? We would have to introduce a new `all` event just for that, which would have an even bigger impact on the API.

As for assumptions, these functions are dedicated to work with Emmitters. That's the one and only assumption and it's by design (and part of the documentation).

Finally, this feature will be much probably used mainly inside the MVC API itself, which is a place we have full control.

We may step back and go with `listenTo()` as well. I would be fine. But let's be sure that it's worth undoing the work which is already in place.